airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/ssh/SshTunnel.java

```diff
@@ -119,17 +134,33 @@ public SshTunnel(final JsonNode config,
       if (tunnelMethod.equals(TunnelMethod.SSH_PASSWORD_AUTH)) {
         Preconditions.checkNotNull(tunnelUserPassword);
       }
-      Preconditions.checkNotNull(remoteDatabaseHost);
-      Preconditions.checkArgument(remoteDatabasePort > 0);
+      // must provide either host/port or endpoint
+      Preconditions.checkArgument((hostKey != null && portKey != null) || endPointKey != null);
+      Preconditions.checkArgument((remoteServiceHost != null && remoteServicePort > 0) || remoteServiceUrl != null);
+      if (remoteServiceUrl != null) {
+        URL urlObject = null;
+        try {
+          urlObject = new URL(remoteServiceUrl);
+        } catch (MalformedURLException e) {
+          e.printStackTrace();
```

Comment:
Minor comment, this could also use `AirbyteTraceMessageUtility.emitConfigError(e, message)` to surface the malformed URL and align with the error messages factory PR